mergify_engine/queue/merge_train.py

```diff
@@ -812,14 +840,12 @@ async def update_state(
         ):
             self.checks_ended_timestamp = date.utcnow()
 
-        if checks_conclusion == check_api.Conclusion.FAILURE:
-            for condition in evaluated_queue_rule.conditions.walk():
-                if (
-                    condition.label == constants.CHECKS_TIMEOUT_CONDITION_LABEL
-                    and not condition.match
-                ):
-                    self.has_timed_out = True
-                    break
+        if (
+            self.checks_conclusion != check_api.Conclusion.FAILURE
+            and await self.checks_have_timed_out(date.utcnow())
+        ):
+            self.has_timed_out = True
+            self.checks_conclusion = check_api.Conclusion.FAILURE
```

Comment:
I don't think we should change `self.checks_conclusion`, self.has_timed_out is enough.